src/server/api/bib.ts

```diff
@@ -137,33 +143,37 @@ async function fetchBibSubjectHeadings(bibId: string) {
   }
 }
 
-async function fetchAllBibItems(
+// TODO: Remove this when view_all endpoint in discovery supports query params
+async function fetchAllBibItemsWithQuery(
   bibQuery: BibQueryParams,
   numItems: number,
   batchSize: number
 ): Promise<DiscoveryItemResult[]> {
   const items: DiscoveryItemResult[] = []
   const client = await nyplApiClient({ apiName: DISCOVERY_API_NAME })
-  for (
-    let batchNum = 1;
-    batchNum <= Math.ceil(numItems / batchSize);
-    batchNum++
-  ) {
+  const totalBatchNum = Math.ceil(numItems / batchSize)
+
+  for (let batchNum = 1; batchNum <= totalBatchNum; batchNum++) {
     const pageQueryString = getBibQueryString(
       {
         ...bibQuery,
         item_page: batchNum,
       },
-      false,
-      true
+      false
     )
     const bibPage = await client.get(
       `${DISCOVERY_API_SEARCH_ROUTE}/${bibQuery.id}${pageQueryString}`
     )
-    if (bibPage?.items?.length) {
-      items.push(...bibPage.items)
-    } else {
-      throw new Error("There was en error fetching items in one of the batches")
+    try {
```

Comment:
The `client.get` call should go in the try statement. Because if that call fails and throws an error, it's still not caught. What's being tested here is if `bibPage.items` has values but that wouldn't really crash.